### PR TITLE
Fixes #30950 - Enable SmartProxy Registration module

### PR DESCRIPTION
**Pull request by stejskalleos**
Waiting for: https://github.com/theforeman/smart-proxy/pull/768

---

PR is part of **Simple & automatic host registration WF**, see:
* [RFC](https://community.theforeman.org/t/rfc-simple-automatic-host-registration-wf/19588)
* [Redmine (this PR)](https://projects.theforeman.org/issues/30950)
* [Redmine (Parent task)](https://projects.theforeman.org/issues/30440)

### CODE REVIEW

File: manifests/module/registration.pp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# @summary The built in Registration module
+#
+# @param enabled
+#  Whether the module is enabled or disabled.
+#
+# @param listen_on
+#   Where to listen on.
+class foreman_proxy::module::registration (
+  Boolean $enabled = $foreman_proxy::registration,
+  Foreman_proxy::ListenOn $listen_on = $foreman_proxy::registration_listen_on,
```
Comment (ekohl) [inline manifests/module/registration.pp:10]: These need to exist in `init.pp`

**Review by ekohl — CHANGES_REQUESTED** [review]
Please also write in the compatibility section in README.md that users on older versions should disable the module since it doesn't exist.

You may also need to create a template for the settings.

Acceptance tests will fail until it's merged in the smart proxy, but at least the unit tests can pass before that.

File: manifests/init.pp [hunk 2]
```diff
@@ -361,6 +361,8 @@
   Boolean $templates = $foreman_proxy::params::templates,
   Foreman_proxy::ListenOn $templates_listen_on = $foreman_proxy::params::templates_listen_on,
   Stdlib::HTTPUrl $template_url = $foreman_proxy::params::template_url,
+  Boolean $registration = $foreman_proxy::params::registration,
+  Foreman_proxy::ListenOn $registration_listen_on = $foreman_proxy::params::registration_listen_on,
```
Comment (ekohl) [inline manifests/init.pp:365]: If you refer to variables in params, the need to be defined in params.pp as well.

File: manifests/params.pp [hunk 3]
```diff
@@ -229,6 +229,10 @@
   $templates_listen_on = 'both'
   $template_url        = "http://${facts['networking']['fqdn']}:${http_port}"
 
+  # Registration settings
+  $registration           = true
+  $registration_listen_on = 'both'
```
Comment (ekohl) [inline manifests/params.pp:234]: Do we need this to be available on HTTP? Aren't we trying to get HTTPS working to make it secure?

File: manifests/module/registration.pp [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+) {
+  foreman_proxy::module { 'registration':
+    enabled   => $enabled,
+    feature   => 'Registration',
```
Comment (ekohl) [inline manifests/module/registration.pp:14]: Note to self: confirmed that this matches Foreman's feature name (https://github.com/theforeman/foreman/commit/31b0d198080ca9f1444308f726870f0bf49208ec)

**Review by ekohl — COMMENTED** [review]
Overall it looks good. Just wondering if we should enforce HTTPS-only for security.

**Comment by ares** [comment]
After a quick clarification, the pending thing here is to enforce https for this module. Then should be good to go.

**Review by ekohl — APPROVED** [review]

**Pull request merged**